src/util/aws-util.ts

```diff
@@ -81,28 +82,25 @@ export class AwsUtil {
      *
      * If profile is provided, then partition will be ignored.
      */
-    public static async Initialize(): Promise<void> {
-        let stsClient = new STSClient();
-        // running with a profile in the commercial region
-        if (this.profile) {
-            process.env.AWS_SDK_LOAD_CONFIG = '1';
-            process.env.AWS_PROFILE = this.profile;
-            const caller = await stsClient.send(new GetCallerIdentityCommand({}));
-            this.userId = caller.UserId.replace(':', '-').substring(0, 60);
-            return;
-        }
-        // targeting a partition
-        if (this.isPartition) {
-            // if targeting partition with a profile
-            if (this.partitionProfile) {
-                process.env.AWS_SDK_LOAD_CONFIG = '1';
-                process.env.AWS_PROFILE = this.partitionProfile;
-            }
-            stsClient = new STSClient({ region: this.partitionRegion });
-        }
+    public static async Initialize(credentials?: Provider<AwsCredentialIdentity>): Promise<void> {
+
+        // if no credentials(provider) was passed in this method, we use the default
+        const provider = credentials ?? defaultProvider({
+            profile: this.isPartition ? this.partitionProfile : this.profile,
+        });
+
+        // save the credentialsProvider for any of the services we initialize
+        AwsUtil.credentialsProvider = provider;
+
+        // oc: not sure if this is needed, my guess is not
```

Comment:
I'm not sure. Isn't it necessary to pick up sso credentials for example?

> To do this, specify a credential process in the shared AWS conﬁg file or the shared credentials file. If the AWS_SDK_LOAD_CONFIG environment variable is set to a truthy value, the SDK will prefer the process specified in the config file over the process specified in the credentials file (if any).

Reply:
hmm.. the default credentials provider seems to be capable of dealing with all of it. we could just try to not set it and see what happens